src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -72,6 +89,12 @@ public sealed class RestoreManagerPackage : AsyncPackage
                 _buildEvents = dte.Events.BuildEvents;
                 _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
 
+                var solutionBuildManager = (IVsSolutionBuildManager5)await GetServiceAsync(typeof(SVsSolutionBuildManager));
+                Assumes.Present(solutionBuildManager);
+
+                uint updateSolutionEventsCookie4;
+                solutionBuildManager.AdviseUpdateSolutionEvents4(this, out updateSolutionEventsCookie4);
```

Comment:
I'd still prefer to see this output the cookie to a field rather than a local so you can unadvise in your package Close/Dispose method.